### ISSUE TITLE
LaTeX: Keep OriginalVerbatim environment for compatibility

### ISSUE THREAD

**Issue by mgeier** [sphinx-doc/sphinx#2698]
In #2652, the LaTeX environment `OriginalVerbatim` was (re-)renamed to `Verbatim`, which is good.
However, this breaks [my extension](http://nbsphinx.readthedocs.io/).

Could you please keep the definition of `OriginalVerbatim` for backwards compatibility?

**Comment by jfbu** [comment]
hi, sorry about that. If I understand correctly you need back

```
\let\OriginalVerbatim=\Verbatim
\let\endOriginalVerbatim=\endVerbatim
```

or there is some other issue ? (you can test by adding this to `'preamble'` key of `latex_elements`).

Looking at https://github.com/spatialaudio/nbsphinx/blob/master/nbsphinx.py I see you are already preparing for Sphinx 1.5 transition. Thus I guess you have a problem with making the code work on a checkout of current Sphinx master branch ?

**Comment by jfbu** [comment]
shot in the dark here, you have

```
 elif lines[-2].startswith(r'\end{Verbatim}'):  # Sphinx < 1.5
```

but if the code block is in a table Sphinx 1.5 will use Verbatim. The you ask 

```
        out.append(lines[-2].replace('Verbatim', 'OriginalVerbatim'))
```

for `\OriginalVerbatim` which currently has been removed from master branch (leading to future 1.5). I this problem you have or something else?

**Comment by mgeier** [comment]
Yes, the two lines you mentioned would be enough.
It would be great if you could add them back in!
Perhaps with a comment mentioning backwards-compatibility ...

The code you mention is not the problem.
In this case it's easy to check if Sphinx produces `sphinxVerbatim` (>= 1.5) or `Verbatim` (< 1.5) and I can easily restore the non-Sphinx-modified (the "original") environment.

The problem is actually in the Jinja template I'm using (which also contains two occurrences of `OriginalVerbatim`).
In this case I would have to explicitly check for the Sphinx version, because AFAIK there is no other way to find out if `Verbatim` or `OriginalVerbatim` should be used.

**Comment by jfbu** [comment]
I tested on a simple jpynb your current nbsphinx.py with Sphinx master, and it worked fine, but that was probably too simple a case as I see the `OriginalVerbatim` in the `RST_TEMPLATE`. 

I understand this may not be convenient but you could add to your `LATEX_PREAMBLE`:

```
\ifdefined\OriginalVerbatim\else
    % Sphinx >= 1.5, Verbatim is the original one
    \let\OriginalVerbatim\Verbatim \let\endOriginalVerbatim\endVerbatim
\fi
```

**Comment by jfbu** [comment]
anyway, I think it is not big deal for Sphinx 1.5 to still provide `\OriginalVerbatim`,  `\endOriginalVerbatim` even if it does not use it anymore. Hence if above does not work for you, personally I have no strong objections against re-defining them.

**Comment by jfbu** [comment]
fwiw I tested [this ipynb](https://raw.githubusercontent.com/jrjohansson/qutip-lectures/master/Lecture-1-Jaynes-Cumming-model.ipynb) and [your extension](http://nbsphinx.readthedocs.io/) handles it fine except that a problem arises if one activates `sphinx.ext.mathjax`. LaTeX file seems to have nested equations then.

**Comment by mgeier** [comment]
Thanks for checking this out!

Yes, I could use the lines you suggested above for `LATEX_PREAMBLE`, that would be no problem at all.
But it is still a breaking change in Sphinx, and probably someone besides me was using `OriginalVerbatim` before in their own customized LaTeX code?

I don't know if those are relevant, but there are quite a lot of hits on Github: https://github.com/search?q=OriginalVerbatim&type=Code

Either way, I think the right thing to do for Sphinx would be to keep `OriginalVerbatim` for some time for backwards-compatibility (probably somehow marking it as "deprecated"?). After some time has passed and nobody is using Sphinx < 1.5 anymore, it can be dropped.

**Comment by mgeier** [comment]
About the example notebook you mentioned in https://github.com/sphinx-doc/sphinx/issues/2698#issuecomment-227024685:

The `OriginalVerbatim` from the Jinja template is only used in streams (`stdout` and `stderr`) and for exception messages, none of which occur in the notebook.

The thing with the equations is a bug, though.
At first I thought it was because the first two equations are actually in a heading (I guess to make them bigger). But this doesn't seem to be the problem.

As it turns out, the problem is the output of code cell number 13, because it includes an `equation*` environment.
This is actually a known bug in `nbsphinx`: https://github.com/spatialaudio/nbsphinx/pull/44.
I just haven't merged it yet because it exposes a bug in Sphinx < 1.4.2 (#2473).

**Comment by jfbu** [comment]
In PR #2701 the LaTeX writer will again use `OriginalVerbatim` environment and `sphinx.sty` will again define and use `\OriginalVerbatim/\endOriginalVerbatim`. Thus if people customized `OriginalVerbatim` after loading `sphinx.sty` this will be obeyed. People who modified `Verbatim` itself are in no-luck however, as this is now named `sphinxVerbatim`. It is impossible to both not modify fancyvrb Verbatim, and maintain it as synonym for Sphinx custom one. Thanks for having tested out master branch so that we can anticipate problem of future release `1.5` !

**Comment by mgeier** [comment]
Cool, thanks!

**Comment by jfbu** [comment]
by the way I tried  (on Sphinx stable) the example notebook mentioned earlier with your fix spatialaudio/nbsphinx#44 and I confirm it is handled fine for latexpdf by [nbsphinx](http://nbsphinx.readthedocs.io/) also with `sphinx.ext.mathjax`.

**Comment by jfbu** [comment]
but long lines are not wrapped ;-)

**Comment by mgeier** [comment]
Regarding equation line wrapping, is it the problem from https://github.com/jupyter/nbconvert/issues/275?

If you have a solution to that, please tell!

**Comment by jfbu** [comment]
@mgeier ah no sorry that's was only referring to long code lines, not math equations. All I know about that is that LaTeX will break something like this

``` latex
\(\displaystyle
x^{0}+x^{1}+x^{2}+x^{3}+x^{4}+x^{5}+x^{6}+x^{7}+x^{8}+x^{9}+x^{10}+x^{11}+x^{12}+x^{13}+x^{14}+x^{15}+x^{16}+x^{17}+x^{18}+x^{19}+x^{20}+x^{21}+x^{22}+x^{23}+x^{24}+x^{25}+x^{26}+x^{27}+x^{28}+x^{29}+x^{30}+x^{31}+x^{32}+x^{33}+x^{34}+x^{35}+x^{36}+x^{37}+x^{38}+x^{39}+x^{40}+x^{41}+x^{42}+x^{43}+x^{44}+x^{45}+x^{46}+x^{47}+x^{48}+x^{49}+x^{50}+x^{51}+x^{52}+x^{53}+x^{54}+x^{55}+x^{56}+x^{57}+x^{58}+x^{59}+x^{60}+x^{61}+x^{62}+x^{63}+x^{64}+x^{65}+x^{66}+x^{67}+x^{68}+x^{69}+x^{70}+x^{71}+x^{72}+x^{73}+x^{74}+x^{75}+x^{76}+x^{77}+x^{78}+x^{79}+x^{80}+x^{81}+x^{82}+x^{83}+x^{84}+x^{85}+x^{86}+x^{87}+x^{88}+x^{89}+x^{90}+x^{91}+x^{92}+x^{93}+x^{94}+x^{95}+x^{96}+x^{97}+x^{98}+x^{99}+x^{100}+x^{101}+x^{102}+x^{103}+x^{104}+x^{105}+x^{106}+x^{107}+x^{108}+x^{109}+x^{110}+x^{111}+x^{112}+x^{113}+x^{114}+x^{115}+x^{116}+x^{117}+x^{118}+x^{119}\)
```

but not if you use `\[..\]`, `$$..$$` or `equation` environment.

**Comment by mgeier** [comment]
Ah, OK.

IMHO, code shouldn't wrap. I prefer scrollbars for overly long code lines. Even better, the authors should just write shorter lines (according to PEP-8).

**Comment by jfbu** [comment]
@mgeier for latex output and your issue about breaking math, perhaps [autobreak](http://www.ctan.org/pkg/autobreak) will help (just saw the announcement, not tested)

**Comment by jfbu** [comment]
closed via #2701.

memo: OriginalVerbatim will keep being defined and have same meaning it had always; but because of issue #2706, in future rather than using OriginalVerbatim for code-blocks inside tables, PR #2704 has sphinxVerbatimintable instead which knows how to insert the caption and wraps long lines, contrarily to OriginalVerbatim.

**Comment by jfbu** [comment]
@mgeier Thanks for report !
